### PR TITLE
Arc - Unit tests for deleting Postgres

### PR DESCRIPTION
**Pull request by bergeron**
Arc - Adds unit tests for deleting Postgres from the overview page.

Depends on https://github.com/microsoft/azdata-test/pull/2

### CODE REVIEW

File: extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		beforeEach(() => {
+			sinon.stub(utils, 'promptForInstanceDeletion').returns(Promise.resolve(true));
+			sinon.stub(controllerModel, 'acquireAzdataSession').returns(Promise.resolve(vscode.Disposable.from()));
+			refreshTreeNode = sinon.stub(controllerModel, 'refreshTreeNode');
+		});
+
+		it('deletes Postgres on success', async () => {
+			// Stub 'azdata arc postgres server delete' to return success
+			const postgresDelete = sinon.stub();
+			sinon.stub(azdataApi, 'arc').get(() => {
```
Comment (bergeron) [inline extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts:85]: @Charles-Gagnon  I wanted to do:

`sinon.stub(azdataApi.arc.postgres.server, 'delete')`

to stub just the delete function.  It compiles, but didn't override the implementation.  I had to stub the entire arc() getter.

File: extensions/arc/package.json [hunk 2]
```diff
@@ -1130,7 +1130,7 @@
     "@types/sinon": "^9.0.4",
     "@types/uuid": "^8.3.0",
     "@types/yamljs": "^0.2.31",
-    "@microsoft/azdata-test": "^1.4.0",
+    "@microsoft/azdata-test": "file:C:\\azdata-test",
```
Comment (bergeron) [inline extensions/arc/package.json:1133]: @Charles-Gagnon  This requires a new version of azdata-test to include https://github.com/microsoft/azdata-test/pull/2.

File: extensions/arc/src/ui/dashboards/postgres/postgresOverviewPage.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ export class PostgresOverviewPage extends DashboardPage {
 	private properties!: azdata.PropertiesContainerComponent;
 	private kibanaLink!: azdata.HyperlinkComponent;
 	private grafanaLink!: azdata.HyperlinkComponent;
+	private deleteButton!: azdata.ButtonComponent;
```
Comment (bergeron) [inline extensions/arc/src/ui/dashboards/postgres/postgresOverviewPage.ts:33]: Could get the button from the toolbarContainer instead

File: extensions/arc/package.json (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/arc/package.json:1133]: https://www.npmjs.com/package/@microsoft/azdata-test/v/1.5.0

File: extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import { ControllerModel, Registration } from '../../../models/controllerModel';
+
+describe('postgresOverviewPage', () => {
+	let postgresOverview: PostgresOverviewPage;
+	let azdataApi: azdataExt.IAzdataApi;
+	let controllerModel: ControllerModel;
+	let postgresModel: PostgresModel;
+
+	let showInformationMessage: sinon.SinonSpy;
+	let showErrorMessage: sinon.SinonSpy;
```
Comment (bergeron) [inline extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts:29]: `sinon.SinonStub`

File: extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts]: Alright, issue should be fixed. https://github.com/microsoft/azuredatastudio/pull/14502/commits/b2391e52d27ba991496784697244f982536cac2e

The problem was that the fake API was returning a new object each time the arc getting was called - so stubbing arc.<whatever> would stub that object but not the one that the test used.

Fixed it by having there just be one object and returning that instead - which can then be stubbed as needed.

Fixed up the other tests that had this issue as well.

**Review by Charles-Gagnon — APPROVED** [review]
Should be good now assuming they pass the checks

**Review by nasc17 — APPROVED** [review]

File: extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts (hunk 1 above)
Comment (bergeron) [reply extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts]: Ah nice, thanks for the fix

**Pull request merged**